sql/catalyst/src/main/scala-2.13/org/apache/spark/sql/catalyst/expressions/ExpressionSet.scala

```diff
@@ -132,6 +132,12 @@ class ExpressionSet protected(
     newSet
   }
 
+  override def --(that: IterableOnce[Expression]): ExpressionSet = {
+    val newSet = clone()
+    that.iterator.foreach(newSet.remove)
+    newSet
+  }
+
```

Comment:
`++` doesn't have this issue?

Reply:
Yes, ++ no such issue and ExpressionSetSuite already covered ++